amethyst_input/src/input_handler.rs

```diff
@@ -121,13 +118,13 @@ where
                                 .iter()
                                 .cloned(),
                         );
-                        for (k, v) in self.bindings.actions.iter() {
-                            for &button in v {
-                                if Button::Key(key_code) == button {
-                                    event_handler.single_write(ActionReleased(k.clone()));
+                        for (action, combinations) in self.bindings.actions.iter() {
+                            for combination in combinations {
+                                if combination.contains(&Button::Key(key_code)) {
```

Comment:
I'm not super certain this logic works.  I think if you have for example buttons A and B, each with their own action, and then they can also combo with A + B you'll get a release event for the combo if A is pressed and released even if B was never touched.  If I've misread what's happening here please let me know.

Reply:
You missed the next line, the action would be released in that place, so for an action bind to a+b, this will be released when either a or b will be released.

Reply:
Okay now i misread your comment, i will have a look at this later.

Reply:
I just thought of a pretty simple way to resolve this.  Only send this release event if every other button in the combo is currently pressed.

Reply:
Thats pretty much what cames to my mind too while resolving the merge issues and integrating jjedelskys improvements :)